prow/gerrit/client/syncer.go

```diff
@@ -146,25 +150,78 @@ func (st *SyncTime) Update(newState LastSyncState) error {
 		}
 	}
 
-	if !changed {
-		return nil
+	st.val = targetState
+
+	// Pediodically sync time back to storage instead of writing after every
+	// single project is processed, as the I/O would become performance
+	// bottleneck when there are lots of projects.
+	if st.stateChan == nil {
+		st.stateChan = make(chan LastSyncState)
+		go func() {
+			ticker := time.NewTicker(st.uploadInterval)
+			defer ticker.Stop()
+
+			var curState *LastSyncState
+			for {
+				select {
+				case newState := <-st.stateChan:
+					curState = &newState
+				case <-st.forceFlushChan:
+					if curState == nil {
+						continue
+					}
+					st.uploadOnce(*curState)
+					curState = nil
+				case <-ticker.C:
+					if curState == nil {
+						continue
+					}
+					st.uploadOnce(*curState)
+					curState = nil
```

Comment:
How come there are two cases here when they both have the same case body? Maybe collapse these two cases (use `fallthrough` keyword) and then add a comment saying why we need to cater to both the forceFlushChan and the ticker.C channels?

Reply:
`fallthrough` is for switch not select, added comment explaining more about force

Reply:
> fallthrough is for switch not select

Oops, makes sense.